### PR TITLE
#184637261 ft update/delete product 184637261

### PR DESCRIPTION
**Pull request by Jmukakalisa**
**What does this PR do?**

This PR adds a feature that allows sellers to to mark product available/unavailable in their collection

**Description of Task to be completed?**
A seller should be able to update & delete a specific item from their collection

How should this be manually tested?

- clone the repository
- checkout ft-ft-update/delete-product-184637261
- Run npm install to install the dependencies and dev dependencies in your terminal
- Run npm run dev to start the server in terminal and check if server is working
- In URL field enter localhost:PORT/api-docs to open swagger UI
- Login users using /api/v1/users/signin endpoint
- Verify your toke with otp at /api/v1/users/otp/verify/{token} endpoint
- Authorize /api/v1/products/{id} using your verified token in the authorization key
- update a product available/unavailable using /api/v1/products/{id} endpoint
- delete a product available/unavailable using /api/v1/products/{id}/delete endpoint

**Any background context you want to provide?**
none
**What are the relevant pivotal tracker/Trello stories?**
tracker : #184637261
Screenshots (if appropriate)
none
Questions:
none

### CODE REVIEW

**Review by dean-daryl — APPROVED** [review]

**Review by masterchief-00 — APPROVED** [review]

File: src/controllers/productController.js [hunk 1]
```diff
@@ -50,4 +53,114 @@ const CreateProduct = async (req, res) => {
   }
 };
 
+export const updateProductAvailability = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const product = await Product.findOne({ where: { id } });
+    if (!product) {
+      return res.status(400).json({ status: 400, success: false, message: 'Product not found' });
```
Comment (fab-ryan) [inline src/controllers/productController.js]: 404 status not 400

File: src/controllers/productController.js [hunk 2]
```diff
@@ -50,4 +53,114 @@ const CreateProduct = async (req, res) => {
...
+};
+
+export const updateProduct = async (req, res) => {
+  try {
+    const updateData = req.body;
+    if (!Object.keys(updateData).length) {
+      return res.status(400).json({
+        status: 400,
+        success: false,
+        message: 'No data provided',
```
Comment (fab-ryan) [inline src/controllers/productController.js]: same as here

File: src/controllers/productController.js [hunk 3]
```diff
@@ -50,4 +53,114 @@ const CreateProduct = async (req, res) => {
...
+      });
+    }
+    const { id } = req.params;
+    const product = await Product.findOne({ where: { id } });
+    if (!product) {
+      return res.status(400).json({ status: 400, success: false, message: 'Product not found in your collection!' });
```
Comment (fab-ryan) [inline src/controllers/productController.js]: please check all status codes if are correct

File: src/controllers/productController.js [hunk 4]
```diff
@@ -50,4 +53,114 @@ const CreateProduct = async (req, res) => {
...
+    const updatedProduct = await product.update(updateData, { returning: true });
+    if (req.files) {
+      const promises = req.files.map((file) => cloudinary.uploader.upload(file.path, {
+        folder: 'Falcons_E-comm_App/ProductImages',
+        public_id: `${product.id}_image`,
+      }),);
+
+      const results = await Promise.all(promises);
+      updateData.images = results.map((result) => result.url).filter((url) => url);
+    }
```
Comment (fab-ryan) [inline src/controllers/productController.js]: I'm afraid that's not right you updating the product before saving the image first save the image then use returned  imageLink and update it in db if it's there

Comment (fab-ryan) [reply src/controllers/productController.js]: and remember links are array or json format

**Review by fab-ryan — CHANGES_REQUESTED** [review]
work on those comments then you will inform me after

**Pull request closed without merging**